src/components/BulkMemberManager/ResultsTable/ResultsTable.js

```diff
@@ -61,21 +62,35 @@ const ResultsTable = ({
         )
     }
 
-    const selectedResults = results.filter(({ id }) => isSelected(id))
+    const { noneSelected, allSelected, allPending } = results.reduce(
+        (state, { id }) => {
+            if (isSelected(id)) {
+                state.noneSelected = false
+            } else {
+                state.allSelected = false
+            }
+            if (!isPendingChange(id)) {
+                state.allPending = false
+            }
+            return state
+        },
+        {
+            noneSelected: true,
+            allSelected: true,
+            allPending: true,
+        }
+    )
```

Comment:
The above works and only requires a single iteration through the results, but perhaps it's not the most readable. A more readable alternative which would require more iterations, but won't always have to iterate the full list, could look like this:
```suggestion
    const noneSelected =  results.every(({ id }) => !isSelected(id))
    const allSelected =  results.every(({ id }) => isSelected(id))
    const allPending = results.every(({ id }) => isPendingChange(id))
```

Reply:
I think the proposal here is more readable, and given that we're only processing 10 items at a time, it seems the cost of iterations is currently minimal, but I'm fine with either approach and the existing one might be better in case the lists become longer at some point